### ISSUE TITLE
Serializing of vcard with base64 encoded attribute still fails

### ISSUE THREAD

**Issue by scheibler** [eventable/vobject#39]
The recent vobject release 0.9.3 fixed the base64 encoding error during import of a vcard file. But the
export of that same file still fails with the following error message:

```
Traceback (most recent call last):
  [...]
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/khard-0.11.2-py3.5.egg/khard/carddav_object.py", line 1414, in write_to_file
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/base.py", line 255, in serialize
    return behavior.serialize(self, buf, lineLength, validate)
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/behavior.py", line 162, in serialize
    out = base.defaultSerialize(transformed, buf, lineLength)
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/base.py", line 993, in defaultSerialize
    child.serialize(outbuf, lineLength, validate=False)
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/base.py", line 255, in serialize
    return behavior.serialize(self, buf, lineLength, validate)
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/vcard.py", line 227, in serialize
    VCardTextBehavior.serialize(obj, buf, lineLength, validate)
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/behavior.py", line 162, in serialize
    out = base.defaultSerialize(transformed, buf, lineLength)
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/base.py", line 1001, in defaultSerialize
    obj.behavior.encode(obj)
  File "/home/eric/.virtualenvs/khard/lib/python3.5/site-packages/vobject/vcard.py", line 152, in encode
    line.value = codecs.encode(line.value.encode(coding), "base64").decode("utf-8")
AttributeError: 'bytes' object has no attribute 'encode'
```

**Comment by jelmer** [comment]
Also, on python2: 

[Sun Sep 04 17:07:21.394957 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/base.py", line 255, in serialize
[Sun Sep 04 17:07:21.395001 2016] [wsgi:error] [pid 22520]     return behavior.serialize(self, buf, lineLength, validate)
[Sun Sep 04 17:07:21.395045 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/behavior.py", line 162, in serialize
[Sun Sep 04 17:07:21.395086 2016] [wsgi:error] [pid 22520]     out = base.defaultSerialize(transformed, buf, lineLength)
[Sun Sep 04 17:07:21.395124 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/base.py", line 993, in defaultSerialize
[Sun Sep 04 17:07:21.395160 2016] [wsgi:error] [pid 22520]     child.serialize(outbuf, lineLength, validate=False)
[Sun Sep 04 17:07:21.395197 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/base.py", line 255, in serialize
[Sun Sep 04 17:07:21.395265 2016] [wsgi:error] [pid 22520]     return behavior.serialize(self, buf, lineLength, validate)
[Sun Sep 04 17:07:21.395316 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/vcard.py", line 227, in serialize
[Sun Sep 04 17:07:21.395356 2016] [wsgi:error] [pid 22520]     VCardTextBehavior.serialize(obj, buf, lineLength, validate)
[Sun Sep 04 17:07:21.395397 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/behavior.py", line 162, in serialize
[Sun Sep 04 17:07:21.395439 2016] [wsgi:error] [pid 22520]     out = base.defaultSerialize(transformed, buf, lineLength)
[Sun Sep 04 17:07:21.395480 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/base.py", line 1001, in defaultSerialize
[Sun Sep 04 17:07:21.395524 2016] [wsgi:error] [pid 22520]     obj.behavior.encode(obj)
[Sun Sep 04 17:07:21.395571 2016] [wsgi:error] [pid 22520]   File "/usr/lib/python2.7/dist-packages/vobject/vcard.py", line 152, in encode
[Sun Sep 04 17:07:21.395616 2016] [wsgi:error] [pid 22520]     line.value = codecs.encode(line.value.encode(coding), "base64").decode("utf-8")
[Sun Sep 04 17:07:21.395662 2016] [wsgi:error] [pid 22520] NameError: global name 'coding' is not defined

**Comment by DamienCassou** [comment]
I have the same problem.

**Comment by wpercy** [comment]
resolved by #49
